packages/algob/src/types-input.ts

```diff
@@ -0,0 +1,35 @@
+import * as z from 'zod';
+
+export const AddressSchema = z.string();
+
+export const ASADescriptionSchema = z.object({
+  total: z.number(),
+  decimals: z.number(),
+  defaultFrozen: z.boolean().optional(),
+  unitName: z.string().optional(),
+  url: z.string().optional(),
+  metadataHash: z.string().optional(),
+  note: z.string().optional(),
+  noteb64: z.string().optional(),
+  manager: AddressSchema.optional(),
+  reserve: AddressSchema.optional(),
+  freeze: AddressSchema.optional(),
+  clawback: AddressSchema.optional()
+})
+  .refine(o => ((o.decimals <= 19) && (o.decimals >= 0)), {
+    message: "Decimals must be between 0(non divisible) and 19",
+    path: ['decimals']
+  })
+  .refine(o => (!o.unitName || (o.unitName && (o.unitName.length <= 8))), {
+    message: "Unit name must not be longer than 8 bytes",
+    path: ['unitName']
+  })
+  .refine(o => (!o.url || (o.url && (o.url.length <= 32))), {
+    message: "URL must not be longer than 32 bytes",
+    path: ['url']
+  })
+  .refine(o => (!o.metadataHash || (o.metadataHash && (o.metadataHash.length <= 32))), {
+    message: "Metadata Hash must not be longer than 32 bytes",
+    path: ['metadataHash']
+  });
```

Comment:
Maybe we can have one define with many `if` inside. Same amount big code and maybe even better readability.

Reply:
I can have one large blob but they will return the same objects anyway.

Reply:
It's not possible with this method.
It accepts two parameters and it doesn't do whole object check.

https://github.com/vriad/zod/#custom-validation